src/ir/cfg_builder.cpp

```
@@ -793,7 +803,8 @@ Program Program::from_sequence(const InstructionSeq& inst_seq, const ProgramInfo
                                const ebpf_verifier_options_t& options) {
     // --- Pass: ValidateOptions --------------------------------------------
     options.validate();
-    assert(info.platform != nullptr && "platform must be set before instruction feature validation");
+    // Preserves the platform-non-null invariant for every subsequent pass in this pipeline.
+    assert(info.platform != nullptr && "info.platform must be set before Program::from_sequence");
```

Clang (14.0.6)

[warning] implicit conversion 'const char *' -> bool

(readability-implicit-bool-conversion)